react-components/src/components/RevealToolbar/SlicerButton.tsx

```diff
@@ -18,35 +20,57 @@ type SliceState = {
   bottomRatio: number;
 };
 
-export const SlicerButton = (): ReactElement => {
+type SlicerButtonProps = {
+  storeStateInUrl: boolean;
+};
+
+export const SlicerButton = ({ storeStateInUrl }: SlicerButtonProps): ReactElement => {
   const viewer = useReveal();
+  const { reveal3DResources } = useRevealResources();
+  const urlParam = useUrlStateParam();
+  const { top, bottom } = storeStateInUrl
+    ? urlParam.getSlicerStateFromUrlParam()
+    : { top: 1, bottom: 0 };
   const [sliceActive, setSliceActive] = useState<boolean>(false);
 
   const [sliceState, setSliceState] = useState<SliceState>({
     minHeight: 0,
     maxHeight: 0,
-    topRatio: 1,
-    bottomRatio: 0
+    topRatio: top,
+    bottomRatio: bottom
   });
 
   const { minHeight, maxHeight, topRatio, bottomRatio } = sliceState;
 
-  // Heuristic to increase chance that update is propagated even
-  // if multiple additions/deletions of models occur.
-  const lastModel =
-    viewer.models.length === 0 ? undefined : viewer.models[viewer.models.length - 1];
-
   useEffect(() => {
-    const box = new Box3();
-    viewer.models.forEach((model) => box.union(model.getModelBoundingBox()));
-
-    const newMaxY = box.max.y;
-    const newMinY = box.min.y;
-
-    if (maxHeight !== newMaxY || minHeight !== newMinY) {
-      setSliceState(getNewSliceState(sliceState, newMinY, newMaxY));
-    }
-  }, [viewer, viewer.models.length, lastModel]);
+    const updateSliceState = (): void => {
+      if (reveal3DResources.length === 0 || viewer === undefined) {
+        return;
+      }
+
+      const box = new Box3();
+      reveal3DResources.forEach((model) => box.union(model.getModelBoundingBox()));
+
+      const newMaxY = box.max.y;
+      const newMinY = box.min.y;
+
+      if (maxHeight !== newMaxY || minHeight !== newMinY) {
+        if (storeStateInUrl && (topRatio !== 1 || bottomRatio !== 0)) {
+          viewer.setGlobalClippingPlanes([
+            new Plane(new Vector3(0, 1, 0), -(newMinY + topRatio * (newMaxY - newMinY))),
+            new Plane(new Vector3(0, -1, 0), newMinY + bottomRatio * (newMaxY - newMinY))
+          ]);
+        }
```

Comment:
Function is becoming too big, so would be nice to extract some parts into helper functions.